natiami/assignment4/FindIslands.cpp

```diff
@@ -0,0 +1,45 @@
+#include"FindIslands.h"
+
+FindIslands::FindIslands(){}
+
+FindIslands::FindIslands(int nr_row, int nr_column, vector<vector<bool> >& mapOfTiles) : nr_row(nr_row), nr_column(nr_column), mapOfTiles(mapOfTiles){

```

Comment:
The number of rows and columns can be deduced from the `mapOfTiles` parameter.
Using the actual dimensions of `mapOfTiles` both makes the interface simpler (less constructor arguments) and can be safer (no danger of a mismatch between the `nr_row`, `nr_column` and the actual dimensions of the tiles map.

To be extra safe, you can add a check that the dimensions fit or that `mapOfTiles` have all its rows in the same length (optional).